python/cudf/cudf/core/column/column.py

```diff
@@ -1988,10 +1988,10 @@ def as_column(
         data = Buffer(arbitrary._data.view("|u1"))
         data = build_column(data, dtype=cudf_dtype)
 
-        mask = arbitrary._mask
-        mask = bools_to_mask(as_column(mask).unary_operator("not"))
-
-        data = data.set_mask(mask)
+        pandas_mask = arbitrary._mask
+        if pandas_mask.any():
```

Comment:
Can we avoid calling `.any()` to check if the pandas array has nulls or not?

Reply:
An alternative that comes to my mind is converting to arrow without performing this mask introspection:

```
data = as_column(pa.Array.from_pandas(arbitrary), dtype=dtype)
```

Reply:
I'd benchmark to see if that's much faster?

Reply:
via pyarrow is a tiny bit faster:

```python
# mask introspection:

In [1]: import cudf

In [2]: import pandas as pd

In [3]: df = pd.DataFrame({"a":[10, 11, None]*10000000, "b":["a", "b", None]*10000000, "c":["abcdw", "weere", None]*10000000})

In [4]: %timeit gdf = cudf.DataFrame.from_pandas(df)
1.27 s ± 30.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [5]: df = pd.DataFrame({"a":[10, 11, 12]*10000000, "b":["a", "b", "c"]*10000000, "c":["abcdw", "weere", "hig"]*10000000})

In [6]: %timeit gdf = cudf.DataFrame.from_pandas(df)
1.43 s ± 37.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)



# via pyarrow:

In [1]: import pandas as pd

In [2]: import cudf

In [3]: df = pd.DataFrame({"a":[10, 11, None]*10000000, "b":["a", "b", None]*10000000, "c":["abcdw", "weere", None]*10000000})

In [4]: %timeit gdf = cudf.DataFrame.from_pandas(df)
1.21 s ± 52.5 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [5]: df = pd.DataFrame({"a":[10, 11, 12]*10000000, "b":["a", "b", "c"]*10000000, "c":["abcdw", "weere", "hig"]*10000000})

In [6]: %timeit gdf = cudf.DataFrame.from_pandas(df)
1.38 s ± 33.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
Switched to going via `pyarrow`